semhash/semhash.py

```diff
@@ -312,35 +310,33 @@ def _validate_if_strings(self, records: Sequence[dict[str, str] | str]) -> Seque
         return dict_records
 
     def _validate_filter_budget(
-        self, budget: float | int | None, records: Sequence[dict[str, str] | str], k: int
+        self,
+        budget: float | int,
+        n_records: int,
     ) -> int:
         """
         Validate the filter budget.
 
         :param budget: The budget to validate, either as a percentage (0 to 1) or an absolute number.
-        :param records: The records to validate against.
-        :param k: The number of top-k records to keep.
+        :param n_records: The total number of records.
         :return: The validated budget as an integer.
-        :raises ValueError: If the budget is not within the valid range or less than k.
+        :raises ValueError: If the budget is not within the valid range.
         """
-        if budget is None:
-            budget = 0.9
-
-        if not (0 <= budget <= 1 or 0 <= budget <= len(records)):
-            raise ValueError("Budget must be between 0 and 1 (as a percentage) or between 0 and the number of records.")
-
-        budget = int(budget) if budget > 1 else int(len(records) * budget)
+        # If a float between 0 and 1, interpret as a percentage, otherwise as an absolute value.
+        budget = int(n_records * budget) if isinstance(budget, float) and budget <= 1 else int(budget)
 
-        if budget < k:
-            raise ValueError("Budget must be greater than or equal to the number of top-k records to keep.")
+        if not (0 <= budget <= n_records):
```

Comment:
can the budget really be 0? I'd rewrite this to a positive predicate, it's easier to parse because you explicitly list the failure cases.

```
if budget < 0 or budget > n_records:
```